Compiling serde_with v3.8.1 error[E0283]: type annotations needed

`cargo install --path spectred --bin spectred` does not finish building `spectred` because of error[EO283] (type annotations needed):
```
  Installing spectred v0.3.14 (/usr/local/src/rusty-spectre/spectred)
    Updating crates.io index
     Locking 497 packages to latest compatible versions
      Adding addr2line v0.22.0 (latest: v0.23.0)
      Adding ahash v0.7.8 (latest: v0.8.11)
      Adding async-channel v1.9.0 (latest: v2.3.1)
      Adding async-io v1.13.0 (latest: v2.3.3)
      Adding async-lock v2.8.0 (latest: v3.4.0)
      Adding attohttpc v0.24.1 (latest: v0.28.0)
      Adding axum v0.6.20 (latest: v0.7.5)
      Adding axum-core v0.3.4 (latest: v0.4.3)
      Adding base64 v0.21.7 (latest: v0.22.1)
      Adding bindgen v0.59.2 (latest: v0.69.4)
      Adding bindgen v0.65.1 (latest: v0.69.4)
      Adding bitflags v1.3.2 (latest: v2.5.0)
      Adding borsh v0.9.3 (latest: v1.5.1)
      Adding borsh-derive v0.9.3 (latest: v1.5.1)
      Adding borsh-derive-internal v0.9.3 (latest: v0.10.3)
      Adding borsh-schema-derive-internal v0.9.3 (latest: v0.10.3)
      Adding cfg_aliases v0.1.1 (latest: v0.2.1)
      Adding clap v2.34.0 (latest: v4.5.7)
      Adding convert_case v0.4.0 (latest: v0.6.0)
      Adding destructure_traitobject v0.2.0 (latest: v0.3.0)
      Adding duration-string v0.3.0 (latest: v0.4.0)
      Adding encode_unicode v0.3.6 (latest: v1.0.0)
      Adding enum-primitive-derive v0.2.2 (latest: v0.3.0)
      Adding env_logger v0.9.3 (latest: v0.11.3)
      Adding event-listener v2.5.3 (latest: v5.3.1)
      Adding faster-hex v0.6.1 (latest: v0.9.0)
      Adding fastrand v1.9.0 (latest: v2.1.0)
      Adding fixedbitset v0.4.2 (latest: v0.5.7)
      Adding foreign-types v0.3.2 (latest: v0.5.0)
      Adding foreign-types-shared v0.1.1 (latest: v0.3.1)
      Adding futures-lite v1.13.0 (latest: v2.3.0)
      Adding generic-array v0.14.7 (latest: v1.0.0)
      Adding gimli v0.29.0 (latest: v0.30.0)
      Adding gloo-timers v0.2.6 (latest: v0.3.0)
      Adding h2 v0.3.26 (latest: v0.4.5)
      Adding hashbrown v0.11.2 (latest: v0.14.5)
      Adding hashbrown v0.12.3 (latest: v0.14.5)
      Adding hermit-abi v0.1.19 (latest: v0.4.0)
      Adding hermit-abi v0.3.9 (latest: v0.4.0)
      Adding http v0.2.12 (latest: v1.1.0)
      Adding http-body v0.4.6 (latest: v1.0.0)
      Adding http-range-header v0.3.1 (latest: v0.4.1)
      Adding hyper v0.14.29 (latest: v1.3.1)
      Adding hyper-timeout v0.4.1 (latest: v0.5.1)
      Adding indexmap v1.9.3 (latest: v2.2.6)
      Adding io-lifetimes v1.0.11 (latest: v2.0.3)
      Adding itertools v0.11.0 (latest: v0.13.0)
      Adding itertools v0.12.1 (latest: v0.13.0)
      Adding librocksdb-sys v0.11.0+8.1.1 (latest: v6.20.3)
      Adding linkme v0.2.10 (latest: v0.3.27)
      Adding linkme-impl v0.2.10 (latest: v0.3.27)
      Adding linux-raw-sys v0.3.8 (latest: v0.6.4)
      Adding linux-raw-sys v0.4.14 (latest: v0.6.4)
      Adding local-ip-address v0.5.7 (latest: v0.6.1)
      Adding matchit v0.7.3 (latest: v0.8.2)
      Adding mio v0.8.11 (latest: v1.0.0)
      Adding nix v0.28.0 (latest: v0.29.0)
      Adding ordered-float v2.10.1 (latest: v4.2.0)
      Adding peeking_take_while v0.1.2 (latest: v1.0.0)
      Adding polling v2.8.0 (latest: v3.7.2)
      Adding proc-macro-crate v0.1.5 (latest: v3.1.0)
      Adding rocksdb v0.21.0 (latest: v0.22.0)
      Adding rustc-hash v1.1.0 (latest: v2.0.0)
      Adding rustix v0.37.27 (latest: v0.38.34)
      Adding rustls v0.21.12 (latest: v0.23.10)
      Adding rustls-pemfile v1.0.4 (latest: v2.1.2)
      Adding rustls-webpki v0.101.7 (latest: v0.102.4)
      Adding secp256k1 v0.28.2 (latest: v0.29.0)
      Adding secp256k1-sys v0.9.2 (latest: v0.10.0)
      Adding socket2 v0.4.10 (latest: v0.5.7)
      Adding strsim v0.8.0 (latest: v0.11.1)
      Adding syn v1.0.109 (latest: v2.0.66)
      Adding sync_wrapper v0.1.2 (latest: v1.0.1)
      Adding textwrap v0.11.0 (latest: v0.16.1)
      Adding tokio-rustls v0.24.1 (latest: v0.26.0)
      Adding tokio-tungstenite v0.21.0 (latest: v0.23.1)
      Adding toml v0.5.11 (latest: v0.8.14)
      Adding tonic v0.10.2 (latest: v0.11.0)
      Adding tonic-build v0.10.2 (latest: v0.11.0)
      Adding tower-http v0.4.4 (latest: v0.5.2)
      Adding tungstenite v0.21.0 (latest: v0.23.0)
      Adding uuid v0.8.2 (latest: v1.8.0)
      Adding wasi v0.11.0+wasi-snapshot-preview1 (latest: v0.13.1+wasi-0.2.0)
      Adding which v4.4.2 (latest: v6.0.1)
      Adding windows-core v0.52.0 (latest: v0.57.0)
      Adding windows-sys v0.48.0 (latest: v0.52.0)
      Adding windows-targets v0.48.5 (latest: v0.52.5)
      Adding windows_aarch64_gnullvm v0.48.5 (latest: v0.52.5)
      Adding windows_aarch64_msvc v0.48.5 (latest: v0.52.5)
      Adding windows_i686_gnu v0.48.5 (latest: v0.52.5)
      Adding windows_i686_msvc v0.48.5 (latest: v0.52.5)
      Adding windows_x86_64_gnu v0.48.5 (latest: v0.52.5)
      Adding windows_x86_64_gnullvm v0.48.5 (latest: v0.52.5)
      Adding windows_x86_64_msvc v0.48.5 (latest: v0.52.5)
      Adding workflow-core v0.12.1 (latest: v0.13.2)
      Adding workflow-core-macros v0.12.1 (latest: v0.13.2)
      Adding workflow-log v0.12.1 (latest: v0.13.2)
      Adding workflow-macro-tools v0.12.1 (latest: v0.13.2)
      Adding workflow-panic-hook v0.12.1 (latest: v0.13.2)
      Adding workflow-rpc v0.12.1 (latest: v0.13.2)
      Adding workflow-rpc-macros v0.12.1 (latest: v0.13.2)
      Adding workflow-task v0.12.1 (latest: v0.13.2)
      Adding workflow-task-macros v0.12.1 (latest: v0.13.2)
      Adding workflow-wasm v0.12.1 (latest: v0.13.2)
      Adding workflow-wasm-macros v0.12.1 (latest: v0.13.2)
      Adding workflow-websocket v0.12.1 (latest: v0.13.2)
  Downloaded clap_lex v0.7.1
  Downloaded clang-sys v1.8.1
  Downloaded anstyle-query v1.1.0
  Downloaded darling_macro v0.20.9
  Downloaded sacabase v2.0.0
  Downloaded suffix_array v0.5.0
  Downloaded mimalloc v0.1.42
  Downloaded num-iter v0.1.45
  Downloaded cdivsufsort v2.0.0
  Downloaded multimap v0.10.0
  Downloaded spectrex v0.3.17
  Downloaded utf8parse v0.2.2
  Downloaded clap_derive v4.5.5
  Downloaded igd-next v0.14.3
  Downloaded prost-derive v0.12.6
  Downloaded prost v0.12.6
  Downloaded rc4 v0.1.0
  Downloaded darling v0.20.9
  Downloaded num v0.4.3
  Downloaded local-ip-address v0.5.7
  Downloaded num-rational v0.4.2
  Downloaded num-complex v0.4.6
  Downloaded prost-build v0.12.6
  Downloaded prost-types v0.12.6
  Downloaded clap v4.5.7
  Downloaded xml-rs v0.8.20
  Downloaded prettyplease v0.2.20
  Downloaded darling_core v0.20.9
  Downloaded num-bigint v0.4.5
  Downloaded crossbeam-channel v0.5.13
  Downloaded idna v1.0.1
  Downloaded clap_builder v4.5.7
  Downloaded libmimalloc-sys v0.1.38
  Downloaded lz4-sys v1.9.5
  Downloaded zstd-sys v2.0.11+zstd.1.5.6
  Downloaded petgraph v0.6.5
  Downloaded libz-sys v1.1.18
  Downloaded openssl-src v300.3.1+3.3.1
  Downloaded 38 crates (13.8 MB) in 2.55s (largest was `openssl-src` at 9.8 MB)
   Compiling serde v1.0.203
   Compiling log v0.4.21
   Compiling num-traits v0.2.19
   Compiling jobserver v0.1.31
   Compiling mio v0.8.11
   Compiling tracing-attributes v0.1.27
   Compiling wasm-bindgen v0.2.92
   Compiling generic-array v0.14.7
   Compiling rustix v0.38.34
   Compiling bs58 v0.5.1
   Compiling either v1.12.0
   Compiling prettyplease v0.2.20
   Compiling glob v0.3.1
   Compiling bitflags v2.5.0
   Compiling minimal-lexical v0.2.1
   Compiling wasm-bindgen-backend v0.2.92
   Compiling cc v1.0.99
   Compiling linux-raw-sys v0.4.14
   Compiling libloading v0.8.3
   Compiling bitflags v1.3.2
   Compiling nom v7.1.3
   Compiling rustc-hash v1.1.0
   Compiling peeking_take_while v0.1.2
   Compiling tokio v1.38.0
   Compiling clang-sys v1.8.1
   Compiling lazycell v1.3.0
   Compiling shlex v1.3.0
   Compiling lazy_static v1.4.0
   Compiling itertools v0.12.1
   Compiling bytes v1.6.0
   Compiling hashbrown v0.14.5
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling digest v0.10.7
   Compiling chrono v0.4.38
   Compiling sha2 v0.10.8
   Compiling ordered-float v2.10.1
   Compiling bindgen v0.65.1
   Compiling equivalent v1.0.1
   Compiling wasm-bindgen-macro-support v0.2.92
   Compiling heck v0.5.0
   Compiling indexmap v2.2.6
   Compiling openssl-src v300.3.1+3.3.1
   Compiling fixedbitset v0.4.2
   Compiling fastrand v2.1.0
   Compiling indexmap v1.9.3
   Compiling multimap v0.10.0
   Compiling spin v0.9.8
   Compiling tower-layer v0.3.2
   Compiling spectre-hashes v0.3.14 (/usr/local/src/rusty-spectre/crypto/hashes)
   Compiling secp256k1-sys v0.9.2
   Compiling cexpr v0.6.0
   Compiling ring v0.17.8
   Compiling bzip2-sys v0.1.11+1.0.8
   Compiling lz4-sys v1.9.5
   Compiling prost-derive v0.12.6
   Compiling zstd-sys v2.0.11+zstd.1.5.6
   Compiling wasm-bindgen-macro v0.2.92
   Compiling libz-sys v1.1.18
   Compiling tempfile v3.10.1
   Compiling petgraph v0.6.5
   Compiling openssl-sys v0.9.102
   Compiling serde_json v1.0.117
   Compiling tracing v0.1.40
   Compiling smallvec v1.13.2
   Compiling polling v2.8.0
   Compiling async-io v1.13.0
   Compiling kv-log-macro v1.0.7
   Compiling polling v3.7.2
   Compiling parking_lot_core v0.9.10
   Compiling faster-hex v0.9.0
   Compiling async-io v2.3.3
   Compiling parking_lot v0.12.3
   Compiling js-sys v0.3.69
   Compiling uuid v1.8.0
   Compiling async-global-executor v2.4.1
   Compiling async-std v1.12.0
   Compiling spectre-utils v0.3.14 (/usr/local/src/rusty-spectre/utils)
   Compiling serde-value v0.7.0
   Compiling serde_yaml v0.9.34+deprecated
   Compiling workflow-log v0.12.1
   Compiling tokio-util v0.7.11
   Compiling prost v0.12.6
   Compiling h2 v0.3.26
   Compiling prost-types v0.12.6
   Compiling log4rs v1.3.0
   Compiling icu_normalizer v1.5.0
   Compiling untrusted v0.9.0
   Compiling secp256k1 v0.28.2
   Compiling workflow-wasm-macros v0.12.1
   Compiling web-sys v0.3.69
   Compiling serde-wasm-bindgen v0.6.5
   Compiling wasm-bindgen-futures v0.4.42
   Compiling spectre-core v0.3.14 (/usr/local/src/rusty-spectre/core)
   Compiling spectre-txscript-errors v0.3.14 (/usr/local/src/rusty-spectre/crypto/txscript/errors)
   Compiling librocksdb-sys v0.11.0+8.1.1
   Compiling prost-build v0.12.6
   Compiling idna v1.0.1
   Compiling pin-project-internal v1.1.5
   Compiling hashbrown v0.12.3
   Compiling tonic-build v0.10.2
   Compiling url v2.5.1
   Compiling axum-core v0.3.4
   Compiling rustls v0.21.12
   Compiling sct v0.7.1
   Compiling rustls-webpki v0.101.7
   Compiling pin-project v1.1.5
   Compiling axum v0.6.20
   Compiling tower v0.4.13
   Compiling openssl v0.10.64
   Compiling enum-primitive-derive v0.2.2
   Compiling hyper v0.14.29
   Compiling spectre-wasm-core v0.3.14 (/usr/local/src/rusty-spectre/wasm/core)
   Compiling tokio-io-timeout v1.2.0
   Compiling bincode v1.3.3
   Compiling hex v0.4.3
   Compiling async-stream-impl v0.3.5
   Compiling atty v0.2.14
   Compiling matchit v0.7.3
   Compiling duration-string v0.3.0
   Compiling unicode-width v0.1.13
   Compiling tokio-stream v0.1.15
   Compiling textwrap v0.11.0
   Compiling workflow-panic-hook v0.12.1
   Compiling bindgen v0.59.2
   Compiling vec_map v0.8.2
   Compiling home v0.5.9
   Compiling strsim v0.8.0
   Compiling tokio-rustls v0.24.1
   Compiling humantime v2.1.0
   Compiling http-range-header v0.3.1
   Compiling async-stream v0.3.5
   Compiling termcolor v1.4.1
   Compiling ansi_term v0.12.1
   Compiling tower-http v0.4.4
   Compiling which v4.4.2
   Compiling neli-proc-macros v0.1.3
   Compiling xml-rs v0.8.20
   Compiling spectre-p2p-lib v0.3.14 (/usr/local/src/rusty-spectre/protocol/p2p)
   Compiling env_logger v0.9.3
   Compiling clap v2.34.0
   Compiling attohttpc v0.24.1
   Compiling sha1 v0.10.6
   Compiling workflow-core v0.12.1
   Compiling seqlock v0.2.0
   Compiling cdivsufsort v2.0.0
   Compiling num-integer v0.1.46
   Compiling inout v0.1.3
   Compiling xmltree v0.10.3
   Compiling portable-atomic v1.6.0
   Compiling cipher v0.4.4
   Compiling spectre-grpc-core v0.3.14 (/usr/local/src/rusty-spectre/rpc/grpc/core)
   Compiling sacabase v2.0.0
   Compiling ident_case v1.0.1
   Compiling fnv v1.0.7
   Compiling strsim v0.11.1
   Compiling neli v0.6.4
   Compiling num-bigint v0.4.5
   Compiling darling_core v0.20.9
   Compiling libmimalloc-sys v0.1.38
   Compiling hyper-timeout v0.4.1
   Compiling spectre-utils-tower v0.3.14 (/usr/local/src/rusty-spectre/utils/tower)
   Compiling igd-next v0.14.3
   Compiling local-ip-address v0.5.7
   Compiling workflow-wasm v0.12.1
   Compiling workflow-task v0.12.1
   Compiling crossbeam-epoch v0.9.18
   Compiling utf8parse v0.2.2
   Compiling rayon-core v1.12.1
   Compiling crossbeam-deque v0.8.5
   Compiling anstyle-parse v0.2.4
   Compiling darling_macro v0.20.9
   Compiling num-rational v0.4.2
   Compiling suffix_array v0.5.0
   Compiling spectre-math v0.3.14 (/usr/local/src/rusty-spectre/math)
   Compiling spectre-addresses v0.3.14 (/usr/local/src/rusty-spectre/crypto/addresses)
   Compiling spectre-merkle v0.3.14 (/usr/local/src/rusty-spectre/crypto/merkle)
   Compiling salsa20 v0.10.2
   Compiling rc4 v0.1.0
   Compiling num-iter v0.1.45
   Compiling num-complex v0.4.6
   Compiling anstyle v1.0.7
   Compiling spectre-muhash v0.3.14 (/usr/local/src/rusty-spectre/crypto/muhash)
   Compiling spectre-consensus-core v0.3.14 (/usr/local/src/rusty-spectre/consensus/core)
   Compiling siphasher v1.0.1
   Compiling anstyle-query v1.1.0
   Compiling colorchoice v1.0.1
   Compiling is_terminal_polyfill v1.70.0
   Compiling xxhash-rust v0.8.10
   Compiling anstream v0.6.14
   Compiling num v0.4.3
   Compiling spectrex v0.3.17
   Compiling darling v0.20.9
   Compiling toml_datetime v0.6.6
   Compiling serde_spanned v0.6.6
   Compiling clap_lex v0.7.1
   Compiling tonic v0.10.2
   Compiling clap_builder v4.5.7
   Compiling mimalloc v0.1.42
   Compiling toml_edit v0.22.14
   Compiling serde_with_macros v3.8.1
   Compiling rayon v1.10.0
   Compiling clap_derive v4.5.5
   Compiling spectre-txscript v0.3.14 (/usr/local/src/rusty-spectre/crypto/txscript)
   Compiling spectre-mining-errors v0.3.14 (/usr/local/src/rusty-spectre/mining/errors)
   Compiling workflow-perf-monitor v0.0.2
   Compiling crossbeam-channel v0.5.13
   Compiling toml v0.8.14
   Compiling serde_with v3.8.1
error[E0283]: type annotations needed
  --> protocol/p2p/./src/echo.rs:99:13
   |
99 |         id: Vec::from(Uuid::new_v4().as_ref()),
   |             ^^^ cannot infer type for reference `&_`
   |
   = note: multiple `impl`s satisfying `Vec<u8>: From<&_>` found in the `alloc` crate:
           - impl From<&str> for Vec<u8>;
           - impl<T, N> From<&[T; N]> for Vec<T>
             where the constant `N` has type `usize`, T: Clone;
           - impl<T> From<&[T]> for Vec<T>
             where T: Clone;

error[E0283]: type annotations needed
  --> protocol/p2p/./src/echo.rs:99:38
   |
99 |         id: Vec::from(Uuid::new_v4().as_ref()),
   |                                      ^^^^^^
   |
   = note: multiple `impl`s satisfying `Uuid: AsRef<_>` found in the `uuid` crate:
           - impl AsRef<Uuid> for Uuid;
           - impl AsRef<[u8]> for Uuid;
help: try using a fully qualified path to specify the expected types
   |
99 |         id: Vec::from(<Uuid as AsRef<T>>::as_ref(&Uuid::new_v4())),
   |                       ++++++++++++++++++++++++++++              ~

   Compiling spectre-perf-monitor v0.3.14 (/usr/local/src/rusty-spectre/metrics/perf_monitor)
   Compiling clap v4.5.7
   Compiling spectre-alloc v0.3.14 (/usr/local/src/rusty-spectre/utils/alloc)
For more information about this error, try `rustc --explain E0283`.
error: could not compile `spectre-p2p-lib` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `spectred v0.3.14 (/usr/local/src/rusty-spectre/spectred)`, intermediate artifacts can be found at `/usr/local/src/rusty-spectre/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```
That may not be relevant but I'm on Ubuntu 22.04.

--- a/cli/src/modules/guide.txt
+++ b/cli/src/modules/guide.txt
@@ -1,6 +1,6 @@
 Please note - this is an alpha version of the softeware, not all features are currently functional.
 
-If using a dekstop or a web version of this software, you can use Ctrl+'+' or Ctrl+'-' (Command on MacOS) to
+If using a desktop or a web version of this software, you can use Ctrl+'+' or Ctrl+'-' (Command on MacOS) to
 change the terminal font size.
 
 If using a desktop version, you can use Ctrl+M (Command on MacOS) to bring up metrics.

--- a/components/addressmanager/src/lib.rs
+++ b/components/addressmanager/src/lib.rs
@@ -596,7 +596,7 @@ mod address_store_with_cache {
             // The target uniform distribution
             let target_u_dist = Uniform::new(0.0, (num_of_buckets) as f64).unwrap();
             for _ in 0..num_of_trials {
-                // The weight sampled expected uniform distibution
+                // The weight sampled expected uniform distribution
                 let prioritized_address_distribution = am
                     .lock()
                     .iterate_prioritized_random_addresses(HashSet::new())

--- a/components/consensusmanager/src/lib.rs
+++ b/components/consensusmanager/src/lib.rs
@@ -212,7 +212,7 @@ impl StagingConsensus {
         // Drop `prev` so that deletion below succeeds
         drop(prev);
         // Staging was committed and is now the active consensus so we can delete
-        // any pervious, now inactive, consensus entries
+        // any previous, now inactive, consensus entries
         self.manager.delete_inactive_consensus_entries();
     }
 

--- a/crypto/txscript/src/opcodes/mod.rs
+++ b/crypto/txscript/src/opcodes/mod.rs
@@ -2749,7 +2749,7 @@ mod test {
             (1u64, vec![], false),                                // Case 1: 0 = locktime < txLockTime
             (0x800000, vec![0x7f, 0, 0], false),                  // Case 2: 0 < locktime < txLockTime
             (0x800000, vec![0x7f, 0, 0, 0, 0, 0, 0, 0, 0], true), // Case 3: locktime too big
-            (LOCK_TIME_THRESHOLD * 2, vec![0x7f, 0, 0, 0], true), // Case 4: lock times are inconsistant
+            (LOCK_TIME_THRESHOLD * 2, vec![0x7f, 0, 0, 0], true), // Case 4: lock times are inconsistent
         ] {
             let mut tx = base_tx.clone();
             tx.0.lock_time = tx_lock_time;

--- a/protocol/p2p/src/echo.rs
+++ b/protocol/p2p/src/echo.rs
@@ -96,7 +96,7 @@ fn build_dummy_version_message() -> VersionMessage {
         services: 0,
         timestamp: unix_now() as i64,
         address: None,
-        id: Vec::from(Uuid::new_v4().as_ref()),
+        id: Vec::from(Uuid::new_v4().as_bytes()),
         user_agent: String::new(),
         disable_relay_tx: false,
         subnetwork_id: None,

--- a/wasm/examples/nodejs/javascript/transactions/generator.js
+++ b/wasm/examples/nodejs/javascript/transactions/generator.js
@@ -67,7 +67,7 @@ const { encoding, networkId, address : destinationAddress } = require("../utils"
         // to the change address.
         //
         // If the requested amount is greater than the Spectre
-        // transactoin mass, the Generator will create multiple
+        // transaction mass, the Generator will create multiple
         // transactions where each transaction will forward
         // UTXOs to the change address, until the requested
         // amount is reached.  It will then create a final

